rskj-core/src/main/java/co/rsk/peg/federation/FederationStorageProviderImpl.java

```diff
@@ -208,12 +208,47 @@ public void setPendingFederation(PendingFederation federation) {
         pendingFederation = federation;
     }
 
+    @Override
+    public void clearPendingFederation() {
+        setPendingFederation(null);
+    }
+
+    @Override
+    public Optional<Federation> getProposedFederation(FederationConstants federationConstants, ActivationConfig.ForBlock activations) {
+        if (!activations.isActive(RSKIP419)) {
+            return Optional.empty();
+        }
+
+        if (proposedFederation != null || isProposedFederationSet) {
+            return Optional.ofNullable(proposedFederation);
```

Comment:
If `proposedFederation` is null it will not enter here. So why use `ofNullable`?

Reply:
when we clear the proposed federation, we set it to null. And `isProposedFederationSet` to true. So it will enter here and return a null value